internal/testing/require/require.go

```diff
@@ -89,14 +208,137 @@ func CapturePanic(panics func()) (err error) {
+		return
+	}
+}
+
+func equalsPointer(expected, actual interface{}) bool {
+	expectedV := reflect.ValueOf(expected)
+	if expectedV.Kind() != reflect.Ptr {
+		panic("BUG: expected was not a pointer")
+	}
+	actualV := reflect.ValueOf(actual)
+	if actualV.Kind() != reflect.Ptr {
+		panic("BUG: actual was not a pointer")
+	}
+
+	if t1, t2 := reflect.TypeOf(expectedV), reflect.TypeOf(actualV); t1 != t2 {
+		return false
+	} else {
+		return expected == actual
+	}
+}
+
+// True fails if the actual value wasn't.
+//
+//  * formatWithArgs are optional. When the first is a string that contains '%', it is treated like fmt.Sprintf.
+func True(t TestingT, actual bool, formatWithArgs ...interface{}) {
+	if !actual {
+		fail(t, "expected true, but was false", formatWithArgs...)
+	}
 }
 
-func True(t *testing.T, value bool, msgAndArgs ...interface{}) {
-	require.True(t, value, msgAndArgs)
+// Zero fails if the actual value wasn't.
+//
+//  * formatWithArgs are optional. When the first is a string that contains '%', it is treated like fmt.Sprintf.
+//
+// Note: This isn't precise to numeric types, but we don't care as being more precise is more code and tests.
+func Zero(t TestingT, i interface{}, formatWithArgs ...interface{}) {
+	if i == nil {
+		fail(t, "expected zero, but was nil", formatWithArgs...)
+	}
+	zero := reflect.Zero(reflect.TypeOf(i))
+	if i != zero.Interface() {
+		fail(t, fmt.Sprintf("expected zero, but was %v", i), formatWithArgs...)
+	}
 }
 
-func Zero(t *testing.T, i interface{}, msgAndArgs ...interface{}) {
-	require.Zero(t, i, msgAndArgs)
+// fail tries to treat the formatWithArgs as fmt.Sprintf parameters or joins on space.
+func fail(t TestingT, message string, formatWithArgs ...interface{}) {
+	var failure string
+	if len(formatWithArgs) > 0 {
+		if s, ok := formatWithArgs[0].(string); ok && strings.Contains(s, "%") {
+			failure = fmt.Sprintf(message+": "+s, formatWithArgs[1:]...)
+		} else {
+			var builder strings.Builder
+			builder.WriteString(fmt.Sprintf("%s: %v", message, formatWithArgs[0]))
+			for _, v := range formatWithArgs[1:] {
+				builder.WriteByte(' ')
+				builder.WriteString(fmt.Sprintf("%v", v))
+			}
+			failure = builder.String()
+		}
+	} else {
+		failure = message
+	}
+
+	// Don't write the failStack in our own package!
+	if fs := failStack(); len(fs) > 0 {
+		t.Fatal(failure, "\n", strings.Join(fs, "\n\t\t\t"))
+	} else {
+		t.Fatal(failure)
+	}
+}
+
+// failStack returns the stack leading to the require fail, without test infrastructure.
+//
+// Note: This is similar to assert.CallerInfo in testify
+// Note: This is untested because it is a lot of work to do that. The rationale to punt is this is a test-only internal
+// type which returns optional info. Someone can add tests, but they'd need to do that as an integration test in a
+// different package with something stable line-number-wise.
+func failStack() (fs []string) {
```

Comment:
I verified this manually by making another test fail and verifying that the stack was clickable.